cores/esp8266/WString.cpp

```diff
@@ -159,7 +159,8 @@ unsigned char String::changeBuffer(unsigned int maxStrLen) {
     // Fallthrough to normal allocator
     size_t newSize = (maxStrLen + 16) & (~0xf);
     // Make sure we can fit newsize in the buffer
-    if (newSize > CAPACITY_MAX) {
+    size_t maxCapacity = (1U << sizeof(ptr.cap) * 8) - 1;
```

Comment:
Please be careful when refactoring, the original definition was static constexpr auto. At least make this constexpr.

Reply:
thanks for the advice, @devyte 

i have periodically checked the disassembled binaries using `objdump -d`.
in this case, the same as before.